base/rational.jl

```diff
@@ -564,8 +564,18 @@ end
 
 float(::Type{Rational{T}}) where {T<:Integer} = float(T)
 
-gcd(x::Rational, y::Rational) = unsafe_rational(gcd(x.num, y.num), lcm(x.den, y.den))
-lcm(x::Rational, y::Rational) = unsafe_rational(lcm(x.num, y.num), gcd(x.den, y.den))
+function gcd(x::Rational, y::Rational)
+    if (isinf(x) && !isinf(y)) || (!isinf(x) && isinf(y))
```

Comment:
I think this is more clear:
```suggestion
    isinf(x) != isinf(y)
```